### PR TITLE
Support Mice in iPadOS 13.4+

### PR DESCRIPTION
**Pull request by gw280**
## Description

Adds support for iOS 13.4 mouse support.

## Related Issues

Prior PR is https://github.com/flutter/engine/pull/19965
https://github.com/flutter/flutter/issues/54663
https://github.com/flutter/flutter/issues/55809

## Tests

Tests TBA.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [ ] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Headers/FlutterViewController.h [hunk 1]
```diff
@@ -47,8 +47,13 @@ extern NSNotificationName const FlutterSemanticsUpdateNotification;
  * Dart-related state and asynchronous tasks when navigating back and forth between a
  * FlutterViewController and other `UIViewController`s.
  */
+#ifdef __IPHONE_13_4
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Headers/FlutterViewController.h:50]: You shouldn't need this.  The presence of UIPointerInteractionDelegate is determined by the SDK version used to compile the engine.  We don't support multiple versions of the SDK.  The thing that needs to be guarded is the use of the API at runtime.

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 2]
```diff
@@ -1423,4 +1460,67 @@ - (BOOL)isPresentingViewController {
...
+                         defaultRegion:(UIPointerRegion*)defaultRegion API_AVAILABLE(ios(13.4)) {
+  if (request != nil) {
+    auto packet = std::make_unique<flutter::PointerDataPacket>(1);
+    const CGFloat scale = [UIScreen mainScreen].scale;
+    _mouseState.location = {request.location.x * scale, request.location.y * scale};
+
+    flutter::PointerData pointer_data = [self generatePointerDataForMouse];
+
+    pointer_data.signal_kind = flutter::PointerData::SignalKind::kNone;
+    packet->SetPointerData(0 /* index */, pointer_data);
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm:1490]: The proper style for argument comments is `packet->SetPointerData(/*index=*/0, pointer_data);`  If you use that format they can be verified by static analysis.  https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments

**Review by gaaclarke — COMMENTED** [review]
Code looks good, just a few nits.  What are your plans for testing this?

File: shell/platform/darwin/ios/framework/Headers/FlutterViewController.h (hunk 1 above)
Comment (gw280) [reply shell/platform/darwin/ios/framework/Headers/FlutterViewController.h:50]: Oops, thought I'd caught them all. Will remove.

**Review by gaaclarke — APPROVED** [review]
lgtm, nice

**Pull request merged**